src/Solver/time_stepping/MiniSeisSol.cpp

```diff
@@ -43,10 +43,49 @@
 #include <Kernels/Time.h>
 #include <Kernels/Local.h>
 #include <Monitoring/Stopwatch.h>
+#include "utils/env.h"
 
-void seissol::localIntegration( struct GlobalData* globalData,
-                                initializers::LTS& lts,
-                                initializers::Layer& layer ) {
+#ifdef ACL_DEVICE
+#include <Initializer/BatchRecorders/Recorders.h>
+#include "device.h"
+#endif
+
+seissol::mini::Config seissol::mini::getConfig() {
+  const auto rank = seissol::MPI::mpi.rank();
+  constexpr int numRepeats{10};
+  constexpr int numElements{50000};
+
+  Config config{};
+  utils::Env env{};
+
+  try {
+    config.numRepeats = env.get("SEISSOL_MINI_NUM_REPEATS", numRepeats);
+    if (config.numRepeats < 1) {
+      throw std::runtime_error("expecting a positive integer number");
+    }
+  }
+  catch (std::runtime_error& err) {
+    logWarning(rank) << "failed to read `SEISSOL_MINI_NUM_REPEATS`," << err.what();
+    config.numRepeats = numRepeats;
+  }
+
+  try {
+    config.numElements = env.get("SEISSOL_MINI_NUM_ELEMENTS", numElements);
+    if (config.numElements < 1) {
+      throw std::runtime_error("expecting a positive integer number");
+    }
+  }
+  catch (std::runtime_error& err) {
+    logWarning(rank) << "failed to read `SEISSOL_MINI_NUM_ELEMENTS`," << err.what();
+    config.numElements = numElements;
+  }
+  return config;
+}
```

Comment:
Should we really use environment variables here? IMHO everything should be in one place, so I'd prefer to put these values in `parameters.par` if there is not a big argument for having them as environment variables.

Reply:
We can do it but I would do it in a separate PR. 

There are 2 options in general, namely: 1) read parameters from fortran and then pass it from it (ugly); 2) read from YAML::Node.

The second is better. However, it would required to write a dedicated function (e.g., `readSeiSolMiniFromYaml`). I can see that that we have a couple of such functions. The best way is to do refactoring and put them in one place.    

I've already started prototyping the refactoring but the size of the PR is growing.

I would prefer to separate PRs. What do you think?